### PR TITLE
Less unwraps

### PR DESCRIPTION
**Pull request by joaoantoniocardoso**

### CODE REVIEW

File: src/video/video_source_local.rs [hunk 1]
```diff
@@ -542,26 +589,53 @@ impl VideoSource for VideoSourceLocal {
...
         for camera_path in &cameras_path {
-            let camera = Device::with_path(camera_path).unwrap();
-            let caps = camera.query_caps();
+            let device = match Device::with_path(camera_path) {
+                Ok(device) => device,
+                Err(error) => {
+                    error!("Faield to get device {camera_path:?}: {error:?}");
+                    continue;
+                }
+            };
```
Comment (patrickelectric) [inline src/video/video_source_local.rs:626]:
```suggestion
            let device = Device::with_path(camera_path).or_else(|error| {
                error!("Failed to get device {camera_path:?}: {error:?}");
                Err(error)
            });
```
You can also do it with the others parts

**Review by patrickelectric — CHANGES_REQUESTED** [review]

File: src/video/video_source_local.rs (hunk 1 above)
Comment (joaoantoniocardoso) [reply src/video/video_source_local.rs:626]: hmm, that's actually not compatible -- `or_else` would be good if we wanted to provide another path to get a `Result` of the same type.

File: src/custom/mod.rs [hunk 2]
```diff
@@ -17,7 +17,12 @@ arg_enum! {
 
 pub fn create_default_streams() -> Vec<VideoAndStreamInformation> {
     let default_environment: CustomEnvironment = match cli::manager::default_settings() {
-        Some(value) => CustomEnvironment::from_str(value).unwrap(),
+        Some(value) => {
+            let Ok(vec) = CustomEnvironment::from_str(value) else {
+                return vec![];
+            };
+            vec
```
[the remark this answers is not in the document]
Comment (joaoantoniocardoso) [reply src/custom/mod.rs:24]: That's not compatible, but I put the wrong name in the variable we are extracting, maybe creating confusion... I give you two options:

using ok else:

```suggestion
            let Ok(env) = CustomEnvironment::from_str(value) else {
                return vec![];
            };
            env
```

or using match:

```suggestion
            match CustomEnvironment::from_str(value) {
                Ok(env) => env,
                Err(_) => return vec![],
            }
```

[the remark this answers is not in the document]
Comment (joaoantoniocardoso) [reply src/custom/mod.rs:24]: Looking at the bigger context, we could also nest

```Rust
pub fn create_default_streams() -> Vec<VideoAndStreamInformation> {
    if let Some(value) = cli::manager::default_settings() {
        if let Ok(env) = CustomEnvironment::from_str(value) {
            match env {
                CustomEnvironment::BlueROVUDP => return bluerov::udp(),
                CustomEnvironment::BlueROVRTSP => return bluerov::rtsp(),
            }
        }
    }

    vec![]
}
```

[the remark this answers is not in the document]
Comment (joaoantoniocardoso) [reply src/custom/mod.rs:24]: Feels weird to have the main path of the function nested

[the remark this answers is not in the document]
Comment (joaoantoniocardoso) [reply src/custom/mod.rs:24]: Maybe this:

```Rust
pub fn create_default_streams() -> Vec<VideoAndStreamInformation> {
    match cli::manager::default_settings().map(CustomEnvironment::from_str) {
        Ok(Some(CustomEnvironment::BlueROVUDP)) => bluerov::udp(),
        Ok(Some(CustomEnvironment::BlueROVRTSP)) => bluerov::rtsp(),
        _ => vec![],
    }
}
```

[the remark this answers is not in the document]
Comment (patrickelectric) [reply src/custom/mod.rs:24]: the single match makes more sense to me

[the remark this answers is not in the document]
Comment (joaoantoniocardoso) [reply src/custom/mod.rs]: done

**Review by patrickelectric — APPROVED** [review]
Merge at will

**Pull request merged**